### ISSUE TITLE
CurrentContainerID is empty

### ISSUE THREAD

**Issue by cjdell** [jwilder/docker-gen#189]
My CoreOS machine has recently updated and now my nginx-proxy container has stopped working. It is failing to correctly generate the nginx configuration file. I tracked this down to `CurrentContainerID` being empty in the template generation code. If I hard code it to the current ID then the template will generate the correct configuration and my sites will become accessible now. i.e.:

```
{{ $CurrentContainer := where $ "ID" "a0b57d396e39d93d222928574b0dbf24bde615e78bfea8664827c03c4268aa6d" | first }}
```

By inspection, it appears that the regular expression within `context.go` is no longer matching the cgroup strings within `/proc/self/cgroup`.

For example

```
regex := "/docker/([[:alnum:]]{64})$"
```

does not match the lines

```
9:memory:/system.slice/docker-a0b57d396e39d93d222928574b0dbf24bde615e78bfea8664827c03c4268aa6d.scope
8:cpuset:/system.slice/docker-a0b57d396e39d93d222928574b0dbf24bde615e78bfea8664827c03c4268aa6d.scope
7:perf_event:/system.slice/docker-a0b57d396e39d93d222928574b0dbf24bde615e78bfea8664827c03c4268aa6d.scope
6:devices:/system.slice/docker-a0b57d396e39d93d222928574b0dbf24bde615e78bfea8664827c03c4268aa6d.scope
5:freezer:/system.slice/docker-a0b57d396e39d93d222928574b0dbf24bde615e78bfea8664827c03c4268aa6d.scope
4:blkio:/system.slice/docker-a0b57d396e39d93d222928574b0dbf24bde615e78bfea8664827c03c4268aa6d.scope
3:cpu,cpuacct:/system.slice/docker-a0b57d396e39d93d222928574b0dbf24bde615e78bfea8664827c03c4268aa6d.scope
2:net_cls,net_prio:/system.slice/docker-a0b57d396e39d93d222928574b0dbf24bde615e78bfea8664827c03c4268aa6d.scope
1:name=systemd:/system.slice/docker-a0b57d396e39d93d222928574b0dbf24bde615e78bfea8664827c03c4268aa6d.scope
```

However it does match what the output used to be:

```
6:devices:/docker/3601745b3bd54d9780436faa5f0e4f72bb46231663bb99a6bb892764917832c2
```

I hope this is helpful. I can provide more information if required. Thanks

**Comment by cjdell** [comment]
Just discovered someone has already spotted this issue:
https://github.com/AHelper/docker-gen/commit/49d5392524b8589f80c9f578cbb467e41e07016e

It seems the code would have to be attempted to deal with both formats.

**Comment by AHelper** [comment]
FYI, from looking over nginx-proxy (master), it now uses CurrentContainerID, which will break existing systems.  Looks like it was from jwilder/nginx-proxy#337.

But yes, a more robust way should be used.  

For nginx-proxy, this is used to check if the other containers are accessible based on comparing networks.

**Comment by jwilder** [comment]
Fixed via #186
